blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/RetryNTimes.java

```diff
@@ -0,0 +1,93 @@
+
+    /**
+     * A Datastax {@link RetryPolicy} that is capable of retrying n times
+     * @param readAttempts
+     * @param writeAttempts
+     * @param unavailableAttempts
+     */
+    public RetryNTimes(int readAttempts, int writeAttempts, int unavailableAttempts) {
+        this.readAttempts = readAttempts;
+        this.writeAttempts = writeAttempts;
+        this.unavailableAttempts = unavailableAttempts;
+    }
+
+    @Override
+    public RetryDecision onReadTimeout(Statement stmnt, ConsistencyLevel cl,
+                                       int requiredResponses, int receivedResponses,
+                                       boolean dataReceived, int rTime) {
+        if (dataReceived) {
+            return RetryDecision.ignore();
+        } else if (rTime < readAttempts) {
+            LOG.info(String.format("Retrying on ReadTimeout: stmnt %s, " +
+                                   "consistency %s, requiredResponse %d, " +
+                                   "receivedResponse %d, dataReceived %s, rTime %d",
+                            stmnt, cl, requiredResponses, receivedResponses, dataReceived, rTime));
+            return RetryDecision.retry(cl);
+        } else {
+            return RetryDecision.rethrow();
+        }
+
+    }
+
+    @Override
+    public RetryDecision onWriteTimeout(Statement stmnt, ConsistencyLevel cl,
+                                        WriteType wt, int requiredResponses,
+                                        int receivedResponses, int wTime) {
+        if (wTime < writeAttempts) {
+            LOG.info(String.format("Retrying on WriteTimeout: stmnt %s, " +
+                                   "consistency %s, writeType %s, requiredResponse %d, " +
+                                   "receivedResponse %d, rTime %d",
+                    stmnt, cl, wt.toString(), requiredResponses, receivedResponses, wTime));
+            return RetryDecision.retry(cl);
+        }
+        return RetryDecision.rethrow();
+    }
+
+    @Override
+    public RetryDecision onUnavailable(Statement stmnt, ConsistencyLevel cl,
+                                       int requiredResponses, int receivedResponses, int uTime) {
+        if (uTime < unavailableAttempts) {
+            LOG.info(String.format("Retrying on unavailable: stmnt %s, consistency %s, " +
+                                   "requiredResponse %d, receivedResponse %d, rTime %d",
+                    stmnt, cl, requiredResponses, receivedResponses, uTime));
+            return RetryDecision.retry(ConsistencyLevel.ONE);
+        }
+        return RetryDecision.rethrow();
+    }
+
+    @Override
+    public RetryDecision onRequestError(Statement statement, ConsistencyLevel cl, DriverException e, int nbRetry) {
+        LOG.info(String.format("Trying nextHost on requestError: stmnt %s, consistency %s, driver ex %s, nbRetry %d",
+                statement, cl, e, nbRetry));
+        return RetryDecision.tryNextHost(cl);
+    }
+
+    @Override
```

Comment:
It's a method in the interface, I have to provide an implementation. I guess I can remove the ```@Override``` but I was keeping it the same as the other methods.